resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ResourceTest.java

```diff
@@ -406,8 +407,38 @@ protected void cleanup() throws CoreException {
 			}
 		}, null);
 		getWorkspace().save(true, null);
-		//don't leak builder jobs, since they may affect subsequent tests
+		// don't leak builder jobs, since they may affect subsequent tests
 		waitForBuild();
+		deleteFilesInWorkspace();
+	}
+
+	private void deleteFilesInWorkspace() {
```

Comment:
this deletes files in the Filesystem - not in the workspace. It would be better to remove them with workspace API. Otherwise the workspace will be out of sync with filesystem. Probably there is already helpercode for that somewher.e (isn't that line 402??!)

Reply:
Yes, the code deletes files in the filesystem and not in the workspace. This was my intention, which was obviously not properly reflected by the naming. The reason for that proposal is improper cleanup by different test classes, which lead to the Eclipse workspace being cleaned (by deleting the root), but files still remaining in the file system at the workspace location. In subsequent tests, this could lead to failures due to unexpected files being present.
My proposal was kind of a fallback solution for tests that somehow only clean the workspace but not all files in the workspace directory. However, I agree with you that this may not be the best option. I will figure out _why_ files are still present. One reason I saw is that tests delete the workspace root without forcing files in closed projects to be deleted, but there may also be other cases.

However, I will separate that from this pull request and make a distinct issue/pull request for the file cleanup topic. I have removed the according changes (i.e. the `deleteFilesInWorkspace` method in `ResourceTest`) from the pull request.